solutions/cf-1555B.mdx

```diff
@@ -0,0 +1,330 @@
+	# if ans is float('inf'), there is no way to fit the two rectangles in the room
+	print(ans if ans != float('inf') else -1)
+```
+</PySection>
+<JavaSection>
+
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Main {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio();
+		final int INF = (int) 1e9; //just a really large number
+		int tc = io.nextInt();
+		for(int t = 0; t < tc; t++){
+			int totalWidth = io.nextInt();
+			int totalHeight = io.nextInt();
+			int x1 = io.nextInt();
+			int y1 = io.nextInt(); 
+			int x2 = io.nextInt(); 
+			int y2 = io.nextInt(); 
+
+			int w1 = x2 - x1;
+			int h1 = y2 - y1;
+
+			int w2 = io.nextInt();
+			int h2 = io.nextInt();
+			/*These are the optimal placements to just
+			barely fit the second rectangle */
+
+			// minimum X value needed to fit on left
+			int leftPlace = w2;
+
+			// maximum Y value needed to fit on the right
+			int rightPlace = totalWidth - w2;
+
+			// the maximum Y value needed to fit on top
+			int topPlace = totalHeight - h2;
+
+			// minimum Y value needed to fit on the bottom
+			int bottomPlace = h2;
+
+			/*
+			 *	Going through all possibilities:
+			 *	Top, Bottom, Left, Right
+			 *	Calculate the distance needed to move
+			 *	to the corresponding places.
+				
+			 *	If that distance is negative, that means we don't have
+			 *	to move the initial rectangle
+			 */
+
+			// amount needed to move leftMost position in the rectangle to the leftPlace
+			int amountNeededLeft = Math.max(leftPlace - x1, 0);
+
+			// amount needed to move rightMost position in the rectangle to the rightPlace
+			int amountNeededRight = Math.max(x2 - rightPlace, 0);
+
+			// amount needed to move topMost position in the rectangle to the rightPlace
+			int amountNeededTop = Math.max(y2 - topPlace, 0);
+
+			// amount needed to move bottomMost position in the rectangle to the bottomPlace
+			int amountNeededBottom = Math.max(bottomPlace - y1, 0);
+
+			/*
+			 *	if the width of the two rectangles exceeds the bounds
+			 *	of the large rectangle, we cannot fit the two rectangles
+			 *	on the left or right
+			 */
+			if(w1 + w2 > totalWidth){
+				amountNeededLeft = INF;
+				amountNeededRight = INF;
+			}
+
+			/* 
+			 *	if the height of the two rectangles exceeds
+			 *  the height of the large rectangle, we cannot fit the
+			 *  the two on the top or bottom
+			 *
+			if(h1 + h2 > totalHeight){
+				amountNeededTop = INF;
+				amountNeededBottom = INF;
+			}
+
+			int ans = INF; //if ans == 1e9, then the 2 rectangles cannot fit together
+
+			if(ans > amountNeededLeft) ans = amountNeededLeft;
+			if(ans > amountNeededRight) ans = amountNeededRight;
+			if(ans > amountNeededTop) ans = amountNeededTop;
+			if(ans > amountNeededBottom) ans = amountNeededBottom;
+
+
+			// if the rectangles cannot fit in the bigger rectangles in any way, print -1
+			if(ans == INF){
+				System.out.println(-1);
+			}
+			else{
```

Comment:
move on same line (optional)

`} else {`

Reply:
got it